synapse/tests/test_tools_pushfile.py

```diff
@@ -3,48 +3,39 @@
 import synapse.axon as s_axon
 import synapse.cortex as s_cortex
 import synapse.daemon as s_daemon
+import synapse.neuron as s_neuron
 
 import synapse.tools.pushfile as s_pushfile
 
 class TestPushFile(SynTest):
 
     def test_tools_pushfile(self):
         self.skipLongTest()
-        self.skipTest('Skipping until Axons are sewn back together.')
-        with self.getTestDir() as path:
-
-            visipath = os.path.join(path, 'visi.txt')
+        with self.getAxonCore() as env:
 
+            visipath = os.path.join(env.path, 'visi.txt')
             with open(visipath, 'wb') as fd:
                 fd.write(b'visi')
 
-            outp = self.getTestOutp()
-
             with s_daemon.Daemon() as dmon:
 
-                link = dmon.listen('tcp://127.0.0.1:0/')
-
-                port = link[1].get('port')
-
-                core = s_cortex.openurl('ram:///')
-                dmon.onfini(core.fini)
-
-                axon = s_axon.Axon(os.path.join(path, 'axon00'))
-                dmon.onfini(axon.fini)
+                dmonlink = dmon.listen('tcp://127.0.0.1:0/')
+                dmonport = dmonlink[1].get('port')
+                axonurl = 'tcp://127.0.0.1:%d/axon' % dmonport
+                coreurl = 'tcp://127.0.0.1:%d/core' % dmonport
+                dmon.share('axon', env.axon)
+                dmon.share('core', env.core)
+                env.core.setConfOpt('axon:url', axonurl)
 
-                dmon.share('axon00', axon)
-                dmon.share('core00', core)
-
-                core.setConfOpt('axon:url', 'tcp://127.0.0.1:%d/axon00' % port)
-
-                s_pushfile.main(['--tags', 'foo.bar,baz.faz', 'tcp://127.0.0.1:%d/core00' % port, visipath], outp=outp)
-
-                node = core.getTufoByProp('file:bytes')
+                outp = self.getTestOutp()
+                s_pushfile.main(['--tags', 'foo.bar,baz.faz', coreurl, visipath], outp=outp)
 
+                node = env.core.getTufoByProp('file:bytes')
                 self.eq(node[1].get('file:bytes'), '442f602ecf8230b2a59a44b4f845be27')
                 self.eq(node[1].get('file:bytes:size'), 4)
-
                 self.nn(node[1].get('#foo'))
                 self.nn(node[1].get('#foo.bar'))
                 self.nn(node[1].get('#baz'))
                 self.nn(node[1].get('#baz.faz'))
+
+                # FIXME this test doesn't make sure that the file is in the axon...
```

Comment:
we can use the evn.axon to call wants() and ensure we get a empty tuple back

Reply:
Yep - haven't commited that part yet. You need an axonclient to call wants now, plumbing all that stuff in